### PR TITLE
Transpose -> GEMM issue

### PR DESCRIPTION
**Pull request by mirza-halilcevic**
- Fix issue with gemm not detecting transpose in cases where the transpose does not affect the strides
- Resolves https://github.com/migraphx-benchmark/AMDMIGraphX/issues/183

### CODE REVIEW

File: src/targets/gpu/gemm_impl.cpp [hunk 1]
```diff
@@ -129,7 +129,27 @@ auto rocblas_invoke(F f, Pack p, Ts... xs)
...
+    if(s.transposed())
+    {
+        return s.strides().back() != 1;
+    }
+    else if(s.strides() != s.as_standard().strides())
+    {
+        // If the strides are not standard, check if there's a permutation that produces those
+        // strides
+        auto lens_perm = s.lens();
+        std::sort(lens_perm.begin(), lens_perm.end());
```
Comment (pfultz2) [inline src/targets/gpu/gemm_impl.cpp:143]: Why not call `find_permutation`? That should give the correct permutation without needing the loop.

File: src/targets/gpu/gemm_impl.cpp [hunk 2]
```diff
@@ -22,6 +22,7 @@
  * THE SOFTWARE.
  */
 
+#include "migraphx/permutation.hpp"
```
Comment (pfultz2) [inline src/targets/gpu/gemm_impl.cpp]: Use angle brackets for includes.

File: src/targets/gpu/gemm_impl.cpp (hunk 1 above)
Comment (mirza-halilcevic) [reply src/targets/gpu/gemm_impl.cpp]: It seems like I misunderstood the implementation of `find_permutation` at first glance. I see now the purpose behind it. I changed the implementation accordingly.

File: test/verify/test_gemm_transposeb_detect.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+template <migraphx::shape::type_t DType>
+struct test_gemm_transposeb_detect : verify_program<test_gemm_transposeb_detect<DType>>
+{
+    migraphx::program create_program() const
+    {
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        auto a   = mm->add_parameter("a", migraphx::shape{DType, {2, 2, 1}});
+        auto b   = mm->add_parameter("b", migraphx::shape{DType, {2, 2, 1}});
```
[the remark this answers is not in the document]
Comment (umangyadav) [reply test/verify/test_gemm_transposeb_detect.cpp:38]: You would need to increase `k > 1024` to make it use rocBLAS by default.

[the remark this answers is not in the document]
Comment (mirza-halilcevic) [reply test/verify/test_gemm_transposeb_detect.cpp:38]: `k` needs to be 1 for this test to recreate the issue. Is there another way to force it to use rocBLAS?

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**